substrate/client/db/src/lib.rs

```diff
@@ -2538,6 +2541,10 @@ impl<Block: BlockT> sc_client_api::backend::Backend<Block> for Backend<Block> {
 			self.blockchain.unpin(hash);
 		}
 	}
+
+	fn no_gap(&self, operation: &mut Self::BlockImportOperation) {
```

Comment:
What do you think about adding an `enable_gap(bool)` to the `BlockImportOperation` trait instead?
This might just be a subjective design decision. :-)

Reply:
I think I like it more. I don't like `enable_gap`, `create_gap` seems more appropriate, maybe `allow_gap` would be better though since this parameter only allows the gap to be created, it will not guarantee that one will be created. WDYT?

Reply:
LGTM. Green light from me as soon as pipeline is happy